backend/api/generate_api.sh

```diff
@@ -21,122 +21,15 @@
 
 set -ex
 
-DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
-REPO_ROOT="$DIR/../.."
-VERSION="$(cat $REPO_ROOT/VERSION)"
-if [ -z "$VERSION" ]; then
-    echo "ERROR: $REPO_ROOT/VERSION is empty"
-    exit 1
-fi
-TMP_DIR="$(mktemp -d)"
+cd ../..
+# docker run  -it --rm \
+#     --mount type=bind,source="$(pwd)",target=/app/pipelines \
+#     builder /bin/bash
 
-BAZEL_BINDIR=$(bazel info bazel-bin)
-GENERATED_GO_PROTO_FILES="${BAZEL_BINDIR}/backend/api/api_generated_go_sources/src/github.com/kubeflow/pipelines/backend/api/go_client/*.go"
+# Generate API
+docker run  --interactive --rm \
+    --mount type=bind,source="$(pwd)",target=/go/src/github.com/kubeflow/pipelines \
+    testyd /go/src/github.com/kubeflow/pipelines/backend/api/generator.sh
```

Comment:
I was looking at https://goswagger.io/install.html#for-mac-and-linux-users

It could run the docker image as non-root by

```bash
docker run --rm -it  --user $(id -u):$(id -g) -e GOPATH=$HOME/go:/go -v $HOME:$HOME -w $(pwd) quay.io/goswagger/swagger
```

That sounds like a very clever idea, we use the same user and group id as current user. I haven't tried this on our script yet, but I think it should work.

What do you think?

Reply:
I tried it out but get the following issue: 
```
/usr/bin/protoc: Permission denied
```
will try to solve it.